src/platforms/windows/daemon/windowsfirewall.cpp

```diff
@@ -262,60 +266,38 @@ bool WindowsFirewall::allowTrafficForAppOnAdapter(const QString& exePath,
 
 bool WindowsFirewall::enableKillSwitch(int vpnAdapterIndex,
                                        const InterfaceConfig& config) {
-  logger.log() << "Enableling Killswitch Using Adapter:" << vpnAdapterIndex;
-
-  IPAddressRange v4CatchAll("0.0.0.0", 0, IPAddressRange::IPv4);
-  IPAddressRange v6CatchAll("::0", 0, IPAddressRange::IPv6);
-  if (!config.m_allowedIPAddressRanges.contains(v4CatchAll) &&
-      !config.m_allowedIPAddressRanges.contains(v6CatchAll)) {
-    logger.log() << "Allowed IP is not CatchAll, skip-activating Killswitch";
-    return true;
-  }
-
-  if (!blockAll(LOW_WEIGHT)) {
-    logger.log() << "Ruleset failed: blockAll";
-    disableKillSwitch();
-    return false;
-  };
-  logger.log() << "Ruleset applied: blockAll";
-
-  if (!allowTrafficOfAdapter(vpnAdapterIndex, MED_WEIGHT)) {
-    logger.log() << "Ruleset failed: allowTrafficOfAdapter";
-    disableKillSwitch();
-    return false;
-  }
-  logger.log() << "Ruleset applied: allowTrafficOfAdapter";
+  // Checks if the FW_Rule was enabled succesfully,
+  // disables the whole killswitch and returns false if not.
+  #define FW_OK(rule,name)                               \
+    if(!rule){                                        \
+      logger.log() << "Rule failed:" << name << "\n"; \
+      disableKillSwitch();\
+      return false;\
+    }\
+    logger.log() << "Rule enabled:" << name << "\n";
 
-  if (!allowDHCPTraffic(MED_WEIGHT)) {
-    logger.log() << "Ruleset failed: allowDHCPTraffic";
-    disableKillSwitch();
-    return false;
-  }
-  logger.log() << "Ruleset applied: allowDHCPTraffic";
 
-  if (!allowHyperVTraffic(MED_WEIGHT)) {
-    logger.log() << "Ruleset failed: allowHyperVTraffic";
-    disableKillSwitch();
-    return false;
-  }
-  logger.log() << "Ruleset applied: allowHyperVTraffic";
+  logger.log() << "Enableing Killswitch Using Adapter:" << vpnAdapterIndex;
 
-  if (!allowTrafficForAppOnAll(getCurrentPath(), MAX_WEIGHT)) {
-    logger.log() << "Ruleset failed: allowTrafficForAppOnAll";
-    disableKillSwitch();
-    return false;
-  }
-  logger.log() << "Ruleset applied: allowTrafficForAppOnAll: MozillaVPN";
+  IPAddressRange v4CatchAll("0.0.0.0", 0, IPAddressRange::IPv4);
+  IPAddressRange v6CatchAll("::0", 0, IPAddressRange::IPv6);
+  bool hasCatchAllIP = config.m_allowedIPAddressRanges.contains(v4CatchAll) || config.m_allowedIPAddressRanges.contains(v6CatchAll);
```

Comment:
It's worth pointing out that with local network access disabled, this will result in a default route for IPv6, in which case we will call `blockAll()` regardless of whatever has been configured for IPv4. It seems like it would make more sense to always follow the `blockTrafficTo()` case, since that would also wind up including a default route.